client/src/main/java/org/eclipse/hono/client/impl/HonoClientImpl.java

```diff
@@ -392,51 +392,58 @@ private void reconnect(final Handler<AsyncResult<HonoClient>> connectionHandler,
     @Override
     public Future<MessageConsumer> createTelemetryConsumer(
             final String tenantId,
-            final Consumer<Message> messageConsumer) {
+            final Consumer<Message> messageConsumer,
+            final Handler<AsyncResult<Void>> closeHandler) {
 
         return createConsumer(
                 tenantId,
-                () -> newTelemetryConsumer(tenantId, messageConsumer));
+                () -> newTelemetryConsumer(tenantId, messageConsumer, closeHandler));
     }
 
     private Future<MessageConsumer> newTelemetryConsumer(
             final String tenantId,
-            final Consumer<Message> messageConsumer) {
+            final Consumer<Message> messageConsumer,
+            final Handler<AsyncResult<Void>> closeHandler) {
```

Comment:
it doesn't seem to make much sense to use a `Handler<AsyncResult<Void>>` here instead of just a `Handler<Void>` because the close hook in the consumer cannot fail. IMHO we should therefore use a plain `Handler<Void>` here as well.